endToEndTests/test/query.test.js

```diff
@@ -12,12 +12,15 @@ describe('The /query endpoint', () => {
 
   testCases.forEach(testCase =>
     it('should return data for the test case ' + testCase.testCaseName, async () => {
-      const response = await server
-        .post('/query')
-        .send(testCase.query)
-        .expect(200)
-        .expect('Content-Type', 'application/json');
-      return expect(response.body.queryResult).to.deep.equal(testCase.expectedQueryResult);
+      let response = await server.post('/query').send(testCase.query);
+      try {
+        expect(response.status).to.equal(200);
+        expect(response.header['content-type']).to.equal('application/json');
+      } catch (error) {
+        console.error('Error in response header! Error body:', response.body);
+        throw error;
+      }
+      expect(response.body.queryResult).to.deep.equal(testCase.expectedQueryResult);
```

Comment:
Also, I hope you are okay with these changes, as this helps in debugging failing tests, where before the error message was not displayed, if a successful result was expected.